File: test/fixtures/matlab/post_basic_auth_url_encoded_data.m
%% Web Access using Data Import and Export API
uri = 'http://localhost:28139/api/oauth/token/';
body = 'grant_type=client_credentials';
options = weboptions(...
    'Username', 'foo',...
    'Password', 'bar',...
    'MediaType', 'application/x-www-form-urlencoded'...
);
response = webwrite(uri, body, options);

%% HTTP Interface
import matlab.net.*
import matlab.net.http.*
import matlab.net.http.io.*

header = HeaderField('Content-Type', 'application/x-www-form-urlencoded');
uri = URI('http://localhost:28139/api/oauth/token/');
cred = Credentials('Username', 'foo', 'Password', 'bar');
options = HTTPOptions('Credentials', cred);
body = FormProvider('grant_type', 'client_credentials');
response = RequestMessage('post', header, body).send(uri.EncodedURI, options);

File: test/fixtures/r/post_basic_auth_url_encoded_data.r
library(httr)

headers = c(
  `Content-Type` = "application/x-www-form-urlencoded"
)

data = list(
  grant_type = "client_credentials"
)

res <- httr::POST(url = "http://localhost:28139/api/oauth/token/", httr::add_headers(.headers=headers), body = data, encode = "form", httr::authenticate("foo", "bar"))
